cluster/shard_mapper.go

```diff
@@ -84,42 +108,49 @@ func NewRemoteMaper(nodeID, shardID uint64, stmt string, chunkSize int) *RemoteM
 }
 
 // Open connects to the remote node and starts receiving data.
-func (r *RemoteMapper) Open() error {
-	node, err := r.MetaStore.Node(r.nodeID)
+func (r *RemoteMapper) Open() (err error) {
+	defer func() {
+		if err != nil {
+			r.conn.MarkUnusable()
```

Comment:
We had a bug in the `WriteShard` case where connections were not marked as unusable correctly.  Not sure if that will happen here or not but we needed to pass the connection as a parameter to the defer func.  See: https://github.com/influxdb/influxdb/blob/master/cluster/shard_writer.go#L46